pkg/utils/logger.go

```diff
@@ -0,0 +1,229 @@
+/*
+Copyright 2019 IBM Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+A new function package to manage logging in App Navigator Golang components
+*/
+
+package utils
+
+import (
+        "fmt"
+        "time" 
+        "encoding/json"
+)
+
+/*LogLevel values of LogLevel. LogLevel is what user requests*/
+type LogLevel int 
+const (
+        // LogLevelNone request no trace
+        LogLevelNone  LogLevel = 0  
+        // LogLevelWarning request warning trace
+        LogLevelWarning LogLevel = 1
+        // LogLevelError request error trace
+        LogLevelError LogLevel = 2
+        // LogLevelInfo request info trace
+        LogLevelInfo LogLevel = 3
+        // LogLevelDebug request debug trace
+        LogLevelDebug LogLevel = 4
+        // LogLevelEntry request entry trace
+        LogLevelEntry LogLevel = 5
+        // LogLevelAll request all traces
+	LogLevelAll LogLevel = 6
+)
+
+/*LogType values of LogType. LogType is how code categorizes log message*/
+type LogType int 
+const (
+        // LogTypeEntry entry trace type
+        LogTypeEntry LogType = 0  
+        // LogTypeExit exit trace type
+        LogTypeExit LogType = 1
+        // LogTypeInfo info trace type
+        LogTypeInfo LogType = 2
+        // LogTypeWarning warning trace type
+        LogTypeWarning LogType = 3
+        // LogTypeError error trace type
+        LogTypeError LogType = 4
+        // LogTypeDebug debug trace type
+        LogTypeDebug LogType = 5
+)
+
+/*logType array*/
+var logTypes = [6]string{
+	"entry",
+	"exit",
+	"info",
+	"warning",
+	"error",
+	"debug",
+}
+
+/*logLevel array*/ 
+var logLevels = [7]string{
+        "LogLevelNone",
+        "LogLevelWarning", 
+        "LogLevelError", 
+	"LogLevelInfo", 
+	"LogLevelDebug", 
+	"LogLevelEntry", 
+        "LogLevelAll",       
+} 
+
+//flag to enable json log and default is false
+var enableJSONLog bool = false
```

Comment:
Given that the parameter is passed to the constructor, this field should be part of the logger struct (loggerImpl), not a static variable.